src/styles/_catalog.scss

```diff
@@ -1,22 +1,30 @@
 .catalog {
-  display: flex;
-  flex-wrap: wrap;
+  display: grid;
+  grid-template-columns: repeat(4, $card-width);
+  gap: 48px;
+  justify-content: center;
 
-  max-width: 944px;
-  margin: 0 auto;
+  padding-bottom: 48px;
+
+  @media (max-width: 1023px) {
+    grid-template-columns: repeat(3, $card-width);
+  }
+
+  @media (max-width: 767px) {
+    grid-template-columns: repeat(2, $card-width);
+  }
+  
+  @media (max-width: 487px) {
+    grid-template-columns: $card-width;
+  }
```

Comment:
It is better to use Mobile First approach (nowadays it is more then 80% of mobile traffic) 
[link](https://ishadeed.com/article/the-state-of-mobile-first-and-desktop-first/)
Update your media-queries using `min-width`